### ISSUE TITLE
ansible-lint should return specific failure exit code if no file was linted

### ISSUE THREAD

**Issue by ssbarnea** [ansible-community/ansible-lint#758]
##### Summary
In order to avoid accidental false-positive results where not files are loaded and linted, we need to assure that ansible-lint would return an non-zero exit code when, for various reasons it end-up not loading any file.

##### Issue Type

- Bug Report

##### STEPS TO REPRODUCE
```bash
ansible-lint not-existing-file.yml
EXIT: 0
```
Note that arguments could be paths, and that also the ignore rules do apply. Still, if in the end no files was linted at all, we should return special exit code.

Reusing system `[Errno 2]` (No such file or directory) may not be a bad idea, even if the file exist but it was skipped due to ignore lists, but the long description should be adjusted to clarify the reason.

##### Desired Behaviour
Non zero exit code.

##### Actual Behaviour
Zero exit code, not output at all.

**Comment by webknjaz** [comment]
If we implement this, there should be a CLI option to preserve the old behavior.  Some of POSIX apps use things like `-q`/`--quiet`/`--silent` that also cause the process to not return an error code in similar cases.

**Comment by ssbarnea** [comment]
I think we should distinguish between what we display and what we do. A silent mode that passes is perfectly fine to not have any output. The idea is to assure that default behavior is to return non-zero error code if nothing was processed.

I think that would match behavior from pytest which would return error if no tests where found/run.

This makes me believe that maybe `quiet/silent` modes may not best options to control the behavior of return code without files.

**Comment by webknjaz** [comment]
We can have the option named differently. I've taken these names from `man grep` as an example because I cannot recall where else exactly I saw similar args.

**Comment by ssbarnea** [comment]
Fixed on master, missing file are now an error.
